frontend/public/components/build.tsx

```diff
@@ -61,11 +61,27 @@ export const BuildsDetails: React.SFC<BuildsDetailsProps> = ({obj: build}) => {
   </div>;
 };
 
-const envPath = ['spec', 'strategy', 'sourceStrategy'];
+export const getStrategyType = (strategy) => {
+  switch (strategy.type) {
+    case 'Docker':
+      return 'dockerStrategy';
+    case 'Custom':
+      return 'customStrategy';
+    case 'JenkinsPipeline':
+      return 'jenkinsPipelineStrategy';
+    default:
+      return 'sourceStrategy';
```

Comment:
I'd rather check explicitly for `case 'Source'` and return `null` if `strategy.type` doesn't match any we know about. Then we can hide the Environment tab on `null`.

These are all the types currently, but I'd rather it not fail if a new one is added later.